dali/test/dali_operator_test.h

```diff
@@ -23,30 +23,197 @@
 #include <vector>
 #include <memory>
 #include <map>
+#include <string>
+#include <utility>
+#include "dali/test/operator_argument.h"
+
 
 namespace dali {
 namespace testing {
 
-using Arguments = std::map<ArgumentKey, double>;  // TODO(mszolucha) some generalization boost::any?
+using Arguments = std::map<ArgumentKey, TestOpArg>;
+
+namespace detail {
+
+template<typename Backend>
+std::string BackendStringName() {
+  DALI_ENFORCE(false, "Backend not supported. You may want to write your own specialization");
+}
+
+
+template<>
+inline std::string BackendStringName<CPUBackend>() {
+  return "cpu";
+}
+
+
+template<>
+inline std::string BackendStringName<GPUBackend>() {
+  return "gpu";
+}
+
+
+std::unique_ptr<Pipeline> CreatePipeline(size_t batch_size, size_t num_threads) {
+  return std::unique_ptr<Pipeline>(
+          new Pipeline(static_cast<int>(batch_size), static_cast<int>(num_threads), 0));
+}
+
+
+void AddInputToPipeline(Pipeline &pipeline, const TensorListWrapper &input) {
+  assert(input && input.has_cpu());  // External input works only for CPUBackend
+  const std::string input_name = "input";
+  pipeline.AddExternalInput(input_name);
+  pipeline.SetExternalInput(input_name, *input.get<CPUBackend>());
+}
+
+
+void AddOperatorToPipeline(Pipeline &pipeline, const OpSpec &op_spec) {
```

Comment:
Is it really a function that belongs to `detail`?

Reply:
Done